### PR TITLE
Fix rendering bugs

### PR DESCRIPTION
**Pull request by eigenraven**
- Remove GlStateManager, it's useless in 1.7.10 where every mod calls raw opengl calls and can cause bugs due to wrongly cached state because of this
 - General cleanup of the code
 - Fix the blackhole/weird ambient occlusion bug due to a very wrong implementation of fullbright rendering (it permanently set every block's lightlevel to 15)

Split into commits for easier reviewing

Closes <https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/13102>
Closes <https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/12291>

### CODE REVIEW

**Comment by chochem** [comment]
is it also possible this might fix https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/11977 ?

**Comment by eigenraven** [comment]
Unlikely

File: src/main/java/blockrenderer6343/client/renderer/WorldSceneRenderer.java [hunk 1]
```diff
@@ -242,39 +242,44 @@ protected void drawWorld() {
...
-                        ((BW_GlasBlocks) block).getColor(world.getBlockMetadata(pos.x, pos.y, pos.z))[2] / 255f);
-            } else if (!GT_Renderer_Block.INSTANCE
-                    .renderWorldBlock(world, pos.x, pos.y, pos.z, block, block.getRenderType(), bufferBuilder)) {
-                        bufferBuilder.renderBlockByRenderType(block, pos.x, pos.y, pos.z);
-                    }
-        }
-        if (onRender != null) {
-            onRender.accept(this);
+        try {
+            Tessellator.instance.setBrightness(15 << 20 | 15 << 4);
```
Comment (miozune) [inline src/main/java/blockrenderer6343/client/renderer/WorldSceneRenderer.java:250]: I don't fully understand rendering code, but is this still needed when we override `getLightBrightnessForSkyBlocks`?

Comment (eigenraven) [reply src/main/java/blockrenderer6343/client/renderer/WorldSceneRenderer.java:250]: If you don't set it I think the tesselator doesn't always add it to vertex data, so better to keep it explicitly set to be safe

Comment (miozune) [reply src/main/java/blockrenderer6343/client/renderer/WorldSceneRenderer.java:250]: I see

**Review by miozune — APPROVED** [review]
:lets:

**Pull request merged**